crates/goose-server/src/routes/reply.rs

```diff
@@ -151,6 +175,35 @@ async fn handler(
                             &tx,
                         )
                         .await;
+
+                        // Track failed session
+                        if let Some(manager) = global_telemetry() {
+                            let failed_execution = session_execution
+                                .clone()
+                                .with_result(SessionResult::Error(
+                                    "No provider configured".to_string(),
+                                ))
+                                .with_duration(start_time.elapsed());
+                            let _ = manager.track_session_execution(failed_execution).await;
```

Comment:
from here down you're ignoring the error, maybe log something given this is in the server?